### PR TITLE
refactor: workflow cleanup

### LINKED ISSUE #91: refactor: workflow cleanup
Things to do;
- There's a rule `nh_filter_genome` and another rule `filter_nh_transcriptome`. Use the same ortder for these.
-  Several rules use "premir", but I think given that we use "primary miRNA" annotations, they should probably be "pri_mir".
- The only files that should  go into `finish` are the sorted indexed BAM file from the alignments, the merged count table and the outputs of `bedtools intersect`.
- Some of the rule names are not very descriptive/informative. Improve the rule names.

### PR DESCRIPTION
**Pull request by deliaBlue**
This PR closes #91 .

The changes in this PR are:

- Change all "premir" by "primir" (in rules, output files, scripts and test files)
- Change rule names to make them more meaningful
- Update expected output as some output file names have changed.
- Update `cluster.json` to match rule names and remove unused configuration.

### CODE REVIEW

File: workflow/rules/prepare.smk [hunk 1]
```diff
@@ -357,10 +357,10 @@ rule extend_mirs_annotations:
             ),
             extension=config["extension"]
         ),
-        exteneded_premir=expand(
+        exteneded_primir=expand(
```
Comment (uniqueg) [inline workflow/rules/prepare.smk:360]: Should be `extended_primir`, not `exteneded`

File: workflow/rules/prepare.smk [hunk 2]
```diff
@@ -114,7 +114,7 @@ rule extract_transcriptome_seqs:
 ###############################################################################
 
 
-rule trim_fasta:
+rule trim_fasta_seqs_id:
```
Comment (uniqueg) [inline workflow/rules/prepare.smk:117]: Better `trim_fasta_seq_ids`, no?

**Review by uniqueg — CHANGES_REQUESTED** [review]

**Comment by deliaBlue** [comment]
In this push, some extra changes have been add:

- log files now match the rule name
- Output file names are build to be found together (_i.e_ all the alignments start by `alignments`, all `segemehl` outputs start by `segemehl`, so on and so forth).
- All 4 snakefiles have been lint with `snakefmt -l 80`
- The `{sample}` wildcard has been removed from some output files as they will already be stored in the `sample` directory.

As a suggestion, some script names should be changed by the same reason as [this comment](https://github.com/zavolanlab/mirflowz/pull/98#discussion_r1294593218) was made.

As a detail, rule headers could also be modified (some of them) to be more specific.

File: workflow/rules/prepare.smk [hunk 3]
```diff
@@ -225,16 +227,16 @@ rule get_exons_gtf:
 ###############################################################################
 
 
-rule gtftobed:
+rule exons_gtf_to_bed:
```
Comment (uniqueg) [inline workflow/rules/prepare.smk:230]: Better would be to keep the naming policy and start with a verb, e.g., `convert_`

File: workflow/rules/map.smk [hunk 4]
```diff
@@ -220,17 +228,23 @@ rule cutadapt:
...
+        ),
     output:
-        reads=os.path.join(config["output_dir"], "{sample}", "collapsed.fasta"),
+        reads=os.path.join(
+            config["output_dir"], "{sample}", "reads_collapsed.fasta"
+        ),
     params:
         cluster_log=os.path.join(
-            config["cluster_log"], "fastx_collapser_{sample}.log"
+            config["cluster_log"], "collapse__identical_reads_{sample}.log"
```
Comment (uniqueg) [inline workflow/rules/map.smk:242]: Use only one `_`

File: workflow/rules/map.smk [hunk 5]
```diff
@@ -439,28 +459,28 @@ rule sort_genome_oligomap:
 ###############################################################################
 
 
-rule oligomap_genome_toSAM:
+rule oligomap_genome_to_sam:
```
Comment (uniqueg) [inline workflow/rules/map.smk:462]: Start with a verb, e.g., `comvert_genome_to_sam_oligomap`

File: workflow/rules/map.smk [hunk 6]
```diff
@@ -556,17 +588,21 @@ rule sort_transcriptome_oligomap:
 
 
 ###############################################################################
-### Oligomap transcriptome mapping ouput to SAM
+### Oligomap transcriptome mapping output to SAM
 ###############################################################################
 
 
-rule oligomap_transcriptome_toSAM:
+rule oligomap_transcriptome_to_sam:
```
Comment (uniqueg) [inline workflow/rules/map.smk:595]: As above

**Review by uniqueg — CHANGES_REQUESTED** [review]
A few slight changes for consistent rule names. It seems the file names are still not super consistent, but I didn't comment on that because it would have been too much. Also, it's really not that important.

**Review by uniqueg — APPROVED** [review]

**Pull request merged**